### PR TITLE
Add pre-commit check to sort and remove duplicates from the spelling wordlist

### PR DESCRIPTION
**Pull request by jbampton**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
:heart: -> just  some static check problem. If you can rebase it @jbampton it would be great. I think it was broken master for a while.

**Review by mik-laj — APPROVED** [review]

**Comment by kaxil** [comment]
Static checks fails, can you take a look please @jbampton

File: scripts/ci/pre_commit/pre_commit_sort_spelling_wordlist.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# specific language governing permissions and limitations
+# under the License.
+
+# shellcheck source=scripts/ci/libraries/_script_init.sh
+. "$( dirname "${BASH_SOURCE[0]}" )/../libraries/_script_init.sh"
+
+SPELLING_WORDLIST="${AIRFLOW_SOURCES}/docs/spelling_wordlist.txt"
+readonly SPELLING_WORDLIST
+
+export LC_ALL=C
```
Comment (potiuk) [inline scripts/ci/pre_commit/pre_commit_sort_spelling_wordlist.sh:25]: Oh yeah. That one is crucial to get consistency :)

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Nice!

**Pull request merged**